buildSrc/src/main/kotlin/scripts/infrastructure.gradle.kts

```diff
@@ -13,31 +19,37 @@ tasks.named<Wrapper>("wrapper") {
 
 tasks.register("runUnitTests") {
     description = "Runs all Unit Tests."
-    dependsOn(":app:testDebugUnitTest")
+    dependsOn(":app:test${Default.BUILD_FLAVOR}${Default.BUILD_TYPE}UnitTest")
 }
 
 tasks.register("runAcceptanceTests") {
     description = "Runs all Acceptance Tests in the connected device."
-    dependsOn(":app:connectedAndroidTest")
+    dependsOn(":app:connected${Default.BUILD_FLAVOR}${Default.BUILD_TYPE}AndroidTest")
 }
 
 tasks.register("compileApp") {
-    description = "Compiles the Debug Version of the Wire Android Client."
-    dependsOn(":app:assembleDebug")
+    description = "Compiles the Wire Android Client."
+    dependsOn(":app:assemble${Default.BUILD_FLAVOR}${Default.BUILD_TYPE}")
 }
 
 tasks.register("runApp", Exec::class) {
-    description = "Compiles and runs the Debug Version of the Wire Android Client in the connected device."
-    dependsOn("compileApp", ":app:installDebug")
-    tasks.findByName(":app:installDebug")?.mustRunAfter("compileApp")
+    val compileAppTask = "compileApp"
+    val installAppTask = ":app:install${Default.BUILD_FLAVOR}${Default.BUILD_TYPE}"
+
+    description = "Compiles and runs the Wire Android Client in the connected device."
+    dependsOn(compileAppTask, installAppTask)
+    tasks.findByName(installAppTask)?.mustRunAfter(compileAppTask)
 
     val localProperties = File(project.rootDir, "local.properties")
     if (localProperties.exists()) {
         val properties = Properties()
         localProperties.inputStream().use { properties.load(it) }
         val sdkDir = properties["sdk.dir"]
         val adb = "${sdkDir}/platform-tools/adb"
+        
+        val applicationPackage = "com.wire.android.${Default.BUILD_FLAVOR.toLowerCase()}.${Default.BUILD_TYPE.toLowerCase()}"
+        val launchActivity = "com.wire.android.MainActivity"
```

Comment:
This reference will change when #4 is merged in as shown in [these changes](https://github.com/wireapp/wire-android-reloaded/pull/4/files#diff-c8cc2dd42271f2bf39c6aa81eb6a5529)

Reply:
big ➕ from me :)

isn't there a command to launch the `LAUNCHER` activity w/o giving activity name explicitly?

Reply:
I already check this out on the other PR and commented it. It will break. Unless you want to parse the Manifest, I do not see any other way. And this command is a helper that I find useful especially once we establish which is our starting Activity. 

I use the command line and like to compile and run everything from there. We could perfectly remove this task if necessary.

Reply:
Parsing the ```AndroidManifest``` would add too much complexity for such an insignificant task. The Welcome Activity should not change in time, since Android forces you to define one.

Reply:
parsing the manifest is too much work and not necessary. I can update this script on my PR, no problem